listenbrainz/labs_api/labs/api/mbid_mapping.py

```diff
@@ -0,0 +1,230 @@
+        })
+
+    def names(self):
+        return ("mbid-mapping", "MusicBrainz ID Mapping lookup")
+
+    def inputs(self):
+        return ['[artist_credit_name]', '[recording_name]']
+
+    def introduction(self):
+        return """Given the name of an artist and the name of a recording (track)
+                  this query will attempt to find a suitable match in MusicBrainz."""
+
+    def outputs(self):
+        return ['index', 'artist_credit_arg', 'recording_arg',
+                'artist_credit_name', 'release_name', 'recording_name',
+                'release_mbid', 'recording_mbid', 'artist_credit_id']
+
+    def fetch(self, params, offset=-1, count=-1):
+        """ Main entry point for the query """
+
+        args = []
+        for i, param in enumerate(params):
+            args.append((i, param['[artist_credit_name]'],
+                         param['[recording_name]']))
+
+        results = []
+        for index, artist_credit_name, recording_name in args:
+            hit = self.search(artist_credit_name, recording_name)
+            if hit:
+                hit["artist_credit_arg"] = artist_credit_name
+                hit["recording_arg"] = recording_name
+                hit["index"] = index
+                results.append(hit)
+
+        return results
+
+    def detune_query_string(self, query):
+        """
+            Detune (remove known extra cruft) a metadata field. If a known
+            trailing string exists in the input string, that string and everything
+            after it is removed.
+        """
+
+        for s in ("(", "[", " ft ", " ft. ", " feat ", " feat. ", " featuring "):
+            index = query.find(s)
+            if index >= 0:
+                return query[:index].strip()
+
+        return ""
+
+    def compare(self, artist_credit_name, recording_name, artist_credit_name_hit, recording_name_hit):
+        """
+            Compare the fields, print debug info if turned on, and return edit distance as (a_dist, r_dist)
+        """
+
+        if self.debug:
+            print("Q %-60s %-60s" %
+                  (artist_credit_name[:59], recording_name[:59]))
+            print("H %-60s %-60s" %
+                  (artist_credit_name_hit[:59], recording_name_hit[:59]))
+
+        return (distance(artist_credit_name, artist_credit_name_hit), distance(recording_name, recording_name_hit))
+
+    def evaluate_hit(self, hit, artist_credit_name, recording_name):
+        """
+            Evaluate the given prepared search terms and hit. If the hit doesn't match,
+            attempt to detune it and try again for detuned artist and detuned recording.
+            If the hit is good enough, return it, otherwise return None.
+        """
+        ac_hit = hit['document']['artist_credit_name']
+        r_hit = hit['document']['recording_name']
+
+        ac_detuned = self.detune_query_string(ac_hit)
+        r_detuned = self.detune_query_string(r_hit)
+
+        is_alt = False
```

Comment:
`is_alt` is set here and in the loop but does not seem to be used anywhere.

Reply:
This is for something I had before and no longer used, but then when thinking about this, I will bring it back in more detail and another PR. :)